pdf/core/parser.go

```diff
@@ -1103,43 +1103,39 @@ func (parser *PdfParser) parseXrefStream(xstm *PdfObjectInteger) (*PdfObjectDict
 	return trailerDict, nil
 }
 
-// Parse xref table at the current file position.  Can either be a
-// standard xref table, or an xref stream.
+// Parse xref table at the current file position. Can either be a standard xref
+// table, or an xref stream.
 func (parser *PdfParser) parseXref() (*PdfObjectDictionary, error) {
-	var err error
-	var trailerDict *PdfObjectDictionary
-
-	// Points to xref table or xref stream object?
-	bb, _ := parser.reader.Peek(20)
-	if reIndirectObject.MatchString(string(bb)) {
-		common.Log.Trace("xref points to an object.  Probably xref object")
-		common.Log.Trace("starting with \"%s\"", string(bb))
-		trailerDict, err = parser.parseXrefStream(nil)
-		if err != nil {
-			return nil, err
+	const bufLen = 20
+	bb, _ := parser.reader.Peek(bufLen)
+	for i := 0; i < 2; i++ {
```

Comment:
Would be good to have a short comment above the for loop.  Something like
// Try to parse the xrefs first in the nominal location, second time slightly left of the expected location.